x-pack/plugins/search_playground/public/providers/form_provider.tsx

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { FormProvider as ReactHookFormProvider, useForm } from 'react-hook-form';
+import React, { useEffect } from 'react';
+import { useLoadFieldsByIndices } from '../hooks/use_load_fields_by_indices';
+import { ChatForm, ChatFormFields } from '../types';
+import { useLLMsModels } from '../hooks/use_llms_models';
+
+export const FormProvider: React.FC = ({ children }) => {
+  const models = useLLMsModels();
+  const form = useForm<ChatForm>({
+    defaultValues: {
+      prompt: 'You are an assistant for question-answering tasks.',
+      doc_size: 3,
+      source_fields: {},
+      indices: [],
+      summarization_model: {},
+    },
+  });
+  useLoadFieldsByIndices({ watch: form.watch, setValue: form.setValue, getValues: form.getValues });
+
+  useEffect(() => {
+    const defaultModel = models.find((model) => !model.disabled);
+
+    if (defaultModel) {
+      form.setValue(ChatFormFields.summarizationModel, defaultModel);
+    }
+  }, [form, models]);
```

Comment:
what happens here if they add a new model connector behind the scenes? Does it reset the default model?

Reply:
useLLM modelas should be triggered when they switch to that tab again (part of useQuery mechanic), and will change default model. It might even reset the selected by user model to default one

Reply:
It seems like a corner case and we need to discuss do we want to reflect these changes, like removing all connectors in other tab, or indices, and how does playground should react to it.
<img width="1319" alt="image" src="https://github.com/user-attachments/assets/add0097b-323f-4fce-b550-1a85a7e1f393">

Reply:
can we only update the summarisation model if its null then?

Reply:
fixed in [b1be6b3](https://github.com/elastic/kibana/pull/188278/commits/b1be6b376b503990f113893fd96334191a79df02)

Reply:
I've added check so if condition for showing chat page is not work anymore, then show setup page again. So if user remove connector in other tab he will see setup page here instead of empty model